azurerm/internal/services/network/resource_arm_application_gateway.go

```diff
@@ -3724,3 +3726,20 @@ func flattenApplicationGatewayCustomErrorConfigurations(input *[]network.Applica
 
 	return results
 }
+
+func ApplicationGatewayCustomizeDiff(d *schema.ResourceDiff, _ interface{}) error {
+	tier := d.Get("sku.0.tier").(string)
+	capacity, hasCapacity := d.GetOk("sku.0.capacity")
+
+	if hasCapacity {
+		if (strings.EqualFold(tier, string(network.ApplicationGatewayTierStandard)) || strings.EqualFold(tier, string(network.ApplicationGatewayTierWAF))) && (capacity.(int) < 1 || capacity.(int) > 32) {
```

Comment:
This is looks good, but do you think we can make the error message a bit more clear... maybe something like... WDYT?
```
if (strings.EqualFold(tier, string(network.ApplicationGatewayTierStandardV2)) || strings.EqualFold(tier, string(network.ApplicationGatewayTierWAFV2))) && (capacity.(int) < 1 || capacity.(int) > 125) {
	return fmt.Errorf("The value '%d' exceeds the maximum capacity allowed for a %q V2 SKU, the %q SKU must have a capacity value between 1 and 125", capacity, tier, tier)
} else if (capacity.(int) < 1 || capacity.(int) > 32) {
	return fmt.Errorf("The value '%d' exceeds the maximum capacity allowed for a %q V1 SKU, the %q SKU must have a capacity value between 1 and 32", capacity, tier, tier)
}
```